### ISSUE TITLE
"System.ArgumentException: source area overlaps source bounds"

### ISSUE THREAD

**Issue by MHeasell** [MHeasell/Mappy#8]
Via Bugsnag

reported version 1.0.1.0

```
System.ArgumentException: source area overlaps source bounds
   at Mappy.Collections.GridMethods.Copy<T>(IGrid<T> src, IGrid<T> dest, int sourceX, int sourceY, int destX, int destY, int width, int height) in C:\projects\mappy\Mappy\Collections\GridMethods.cs:line 49
   at Mappy.Operations.CopyAreaOperation<T>.Execute() in C:\projects\mappy\Mappy\Operations\CopyAreaOperation.cs:line 40
   at Mappy.Operations.CompositeOperation.Execute() in C:\projects\mappy\Mappy\Operations\CompositeOperation.cs:line 22
   at Mappy.Operations.CompositeOperation.Execute() in C:\projects\mappy\Mappy\Operations\CompositeOperation.cs:line 22
   at Mappy.Operations.CompositeOperation.Execute() in C:\projects\mappy\Mappy\Operations\CompositeOperation.cs:line 22
   at Mappy.Models.UndoableMapModel.ClearSelection() in C:\projects\mappy\Mappy\Models\UndoableMapModel.cs:line 386
   at Mappy.Services.Dispatcher+<>c.<ClearSelection>b__50_0(UndoableMapModel x) in C:\projects\mappy\Mappy\Services\Dispatcher.cs:line 482
   at Mappy.Maybe<T>.IfSome(Action<T> action) in C:\projects\mappy\Mappy\Maybe.cs:line 161
   at Mappy.Services.Dispatcher.ClearSelection() in C:\projects\mappy\Mappy\Services\Dispatcher.cs:line 482
   at Mappy.Models.MapViewViewModel.MouseDown(Point location) in C:\projects\mappy\Mappy\Models\MapViewViewModel.cs:line 201
   at Mappy.UI.Controls.MapViewPanel.MapViewMouseDown(object sender, MouseEventArgs e) in C:\projects\mappy\Mappy\UI\Controls\MapViewPanel.cs:line 47
   at System.Windows.Forms.Control.OnMouseDown(MouseEventArgs e)
   at Mappy.UI.Controls.LayerView.OnMouseDown(MouseEventArgs e) in C:\projects\mappy\Mappy\UI\Controls\LayerView.cs:line 85
   at System.Windows.Forms.Control.WmMouseDown(ref Message m, MouseButtons button, int clicks)
   at System.Windows.Forms.Control.WndProc(ref Message m)
   at System.Windows.Forms.ScrollableControl.WndProc(ref Message m)
   at System.Windows.Forms.Control+ControlNativeWindow.OnMessage(ref Message m)
   at System.Windows.Forms.Control+ControlNativeWindow.WndProc(ref Message m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, int msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(ref MSG msg)
   at System.Windows.Forms.Application+ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, int reason, int pvLoopData)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoopInner(int reason, ApplicationContext context)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoop(int reason, ApplicationContext context)
   at System.Windows.Forms.Application.Run(Form mainForm)
   at Mappy.Program.Main() in C:\projects\mappy\Mappy\Program.cs:line 79
```

**Comment by MHeasell** [comment]
Seem to have reproduced this. Mappy gets into a weird state when you create a map that's smaller than the window size, then start dragging tiles into it.

**Comment by MHeasell** [comment]
Fixed by 1c8929b6e4bb0d923b79b5c3dd2a5e71f4170525
